test/CMakeLists.txt

```diff
@@ -438,21 +438,16 @@ endforeach()
 #===============================================================================
 # The following tests use odb compare or nccmp to check the difference between the output
 #  file, and the reference copy of what the ouput file should look like.
-# The first rgument to the script, is what the output file type is (netcdf or odb)
+# The first argument to the script, is what the output file type is (netcdf or odb)
 # The second rgument to the script, wrapped in quotes, is the command to execute
 #  the ioda converter.
 # The third argument is the name of the output file to compare (one copy to be
 #  created in testrun/ by the converter, and one copy already present in in
 #  testoutput/)
 #===============================================================================
 
-# Typically, a converter is simply doing a format change (ie, copying data directly
-# from the input file). For these cases, use a tolerance of zero for the
-# iodaconv_comp.sh test.
-#
-# For a converter that is generating new data, use a non-zero tolerance.
-set(IODA_CONV_COMP_TOL_ZERO "0.0")
-set(IODA_CONV_COMP_TOL "0.5e-4")
+# Tolerence (relative percent difference) for comparing floating point numbers.
+set(IODA_CONV_COMP_TOL "1.0e-3")
```

Comment:
@srherbener Isn't this a little large? The old one was still very large. An ulp is basically 1.2e-7.

Reply:
We would need a incredibly large number of floating point operations to get errors that big

Reply:
I agree that this seems like an unnecessarily large tolerance. One thing to keep in mind is that it is a relative percent so 1e-3 represents about 5 significant digits matching. Still seems kind of big however.

One surprising result was that I had to make this 1e-3 to get the tests to pass.

I'm happy to try adjusting this to a smaller value that works, but I'm not sure I can find a much smaller value that works with the existing code and tests.

What do you think is the best way forward? Thanks!

Reply:
@srherbener Hmm I am little bewildered about the tests not passing when this value is smaller. The old tolerances was much smaller. Let me take a look to see what is merged lately and why this fails with say 1.e-5 tolerance. There is perhaps a gotcha somewhere.

Reply:
I was able to build this branch and verify that everything but one test passes with the old tolerance. 5e-5. and an even tighter tolerance of 2e-5. There is a problem with bufr not building properly on my spack-stack-1.5 which causes test_iodaconv_ncep_bufr to fail but that isn't a tolerance problem. Can you reduce this value to one of those above? If you are still getting failures can you give me a list of tests that are failing and on what platform you are testing?